gnomad_qc/v4/create_release/create_release_sites_ht_exomes.py

```diff
@@ -0,0 +1,668 @@
+    test: bool,
+    release_exists: bool,
+) -> hl.Table:
+    """
+    Outer join a list of Hail Tables.
+
+    :param base_table: Dataset to use for interval partitioning.
+    :param tables: List of tables to join.
+    :param new_partition_percent: Percent of base_table partitions used for final
+        release Hail Table.
+    :param test: Whether this is for a test run.
+    :param release_exists: Whether the release HT already exists.
+    :return: Hail Table with datasets joined.
+    """
+    logger.info(
+        "Reading in %s to determine partition intervals for efficient join",
+        base_table,
+    )
+    base_ht_path = get_config()[base_table]["path"]
+    base_ht = hl.read_table(base_ht_path)
+    if test:
+        # Filter to PCSK9 for testing.
+        base_ht = hl.filter_intervals(
+            base_ht,
+            [
+                hl.parse_locus_interval(
+                    "chr1:55039447-55064852", reference_genome="GRCh38"
+                )
+            ],
+        )
+    partition_intervals = base_ht._calculate_new_partitions(
+        base_ht.n_partitions() * new_partition_percent
+    )
+
+    # Reorg list so base table is first.
+    tables.remove(base_table)
+    tables.insert(0, base_table)
+
+    logger.info("Joining datasets: %s...", tables)
+    hts = [
+        get_ht(
+            table,
+            _intervals=partition_intervals,
+            test=test,
+            release_exists=release_exists,
+        )
+        for table in tables
+    ]
+    joined_ht = reduce((lambda joined_ht, ht: joined_ht.join(ht, "left")), hts)
+
+    # Track the dataset we've added as well as the source path.
+    # If release in tables, read in the included datasets json
+    # and update the keys to the path for any new tables
+    included_datasets = {}
+    if "release" in tables:
+        with hl.utils.hadoop_open(
+            included_datasets_json_path(
+                test=test,
+                release_version=hl.eval(get_config()["release"]["ht"].version),
+            )
+        ) as f:
+            included_datasets = json.loads(f.read())
+
+    included_datasets.update(
+        {t: get_config(release_exists=release_exists)[t]["path"] for t in tables}
+    )
+    with hl.utils.hadoop_open(
+        included_datasets_json_path(test=test, release_version=args.version), "w"
+    ) as f:
+        f.write(hl.eval(hl.json(included_datasets)))
+
+    return joined_ht
+
+
+def main(args):
+    """Create release ht."""
+    hl.init(
+        log="/create_release_ht.log",
+        tmp_dir="gs://gnomad-tmp-4day",
+        default_reference="GRCh38",
+    )
+
+    logger.info("Creating release HT...")
+    ht = join_hts(
+        args.base_table,
+        args.tables_for_join,
+        args.new_partition_percent,
+        args.test,
+        args.release_exists,
+    )
+
+    ht = hl.filter_intervals(ht, [hl.parse_locus_interval("chrM")], keep=False)
+    ht = ht.filter(hl.is_defined(ht.filters))
```

Comment:
naive question, but is this to make sure we're removing any AS_lowqual sites?

Reply:
Yup!